desktop/libs/notebook/src/notebook/connectors/trino.py

```diff
@@ -321,6 +321,28 @@ def _get_columns(self, database, table):
     ]
 
 
+  @query_error_handler
+  def explain(self, notebook, snippet):
+    statement = snippet['statement'].rstrip(';')
+    explanation = ''
+
+    if statement:
+      try:
+        query_client = TrinoQuery(self.trino_request, 'USE ' + snippet['database'])
+        query_client.execute()
+        query_client = TrinoQuery(self.trino_request, 'EXPLAIN ' + statement)
+        result = query_client.execute()
+        explanation = result.rows
+      except Exception as e:
+        explanation = str(e)
```

Comment:
Adding error in explanation?

Reply:
Yes, that's what we are doing in the the Hive https://github.com/cloudera/hue/blob/master/desktop/libs/notebook/src/notebook/connectors/hiveserver2.py#L608

Reply:
@JohanAhlen Is UI expecting this for both editor v1 and editor v2?
Just a heads up, we need to improve all this flow then once we pick up this roadmap item. cc @agl29

Reply:
It's a bit odd indeed, @agl29 did you try how the UI behave if you would return a 500 request with the error message instead?